crates/rolldown_common/src/types/member_expr_ref.rs

```diff
@@ -9,4 +10,32 @@ use crate::SymbolRef;
 pub struct MemberExprRef {
   pub object_ref: SymbolRef,
   pub props: Vec<CompactStr>,
+  // Span of the whole member expression
+  pub span: Span,
+}
+
+impl MemberExprRef {
+  pub fn new(object_ref: SymbolRef, props: Vec<CompactStr>, span: Span) -> Self {
+    Self { object_ref, props, span }
+  }
+
+  // #[allow(clippy::manual_map)]: Current code is more readable.
+  #[allow(clippy::manual_map)]
+  pub fn resolved_symbol_ref(
+    &self,
+    resolved_map: &FxHashMap<Span, Option<(SymbolRef, Vec<CompactStr>)>>,
+  ) -> Option<SymbolRef> {
```

Comment:
return `None` need to be handled specially in code splitting and treeshaking. The old code fallback to `object_ref`, which is actually wrong.